### DISCUSSION TITLE
is it possible not to wait for the end of playback to reload the playlist?

### DISCUSSION THREAD

**Discussion by Sebasti3n** [savonet/liquidsoap#1776] [Q&A]
Script :
```
# Musique (list  music)
playlistok = playlist(mode="normal",reload=1,reload_mode="watch","dance.pls")
```
the normal reload mode waits for the end of the current title. Is it possible to cut a title and reload the playlist without waiting?

**Comment by gilou** [comment]
Have you tried the conservative parameter ? (I'm not sure it does that, but if something does, that would be it…)

Otherwise I'd say you'll need to have a sort of container handling that situation…

**Comment by smimram** [comment]
You could try using the `on_reload` parameter, which specifies a function which is called when the playlist has reloaded, in order to skip the current song. The difficulty here is that you need to refer to the source which is being defined, so we use a reference `skipper` to the skip function, which is going to be filled later on. The code:
```
# The function to skip, to be filled later
skipper = ref(fun () -> ())

# Function called when reloading
def on_reload(_)
  f = !skipper
  f()
end

playlistok = playlist(mode="normal",reload=1,reload_mode="watch",on_reload=on_reload,"dance.pls")
skipper := playlistok.skip
```
